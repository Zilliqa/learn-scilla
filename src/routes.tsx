// @ts-ignore
import React, { Suspense, lazy } from 'react';

import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';

import LessonListContainer from './containers/lesson-list-container';
import ChapterContainer from './containers/chapter-container';
const LessonCompleteContainer = lazy(() => import('./containers/lesson-complete-container'));
const AccountContainer = lazy(() => import('./containers/account-container'));

import Spinner from './components/spinner';

export const paths = {
  lessonList: '/',
  lessonComplete: '/lesson-complete/:lesson',
  chapter: '/lesson/:lesson/chapter/:chapter',
  account: '/account'
};

export const RouterNode: React.SFC = () => (
  <Router>
    <Suspense fallback={<Spinner />}>
      <Switch>
        <Route exact={true} path={paths.lessonList} component={LessonListContainer} />
        <Route exact={true} path={paths.chapter} component={ChapterContainer} />
        <Route exact={true} path={paths.lessonComplete} component={LessonCompleteContainer} />
        <Route exact={true} path={paths.account} component={AccountContainer} />
        <Redirect from="/" to={paths.lessonList} />
      </Switch>
    </Suspense>
  </Router>
);
