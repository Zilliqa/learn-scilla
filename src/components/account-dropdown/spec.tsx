import React from 'react';
import AccountDropdown from '.';
import * as renderer from 'react-test-renderer';
import { paths } from '../../routes';

const t = (s: string) => s;
const username = 'noel';
const pathname = '/account';
const logout = () => console.log('logout');
const navigateToAccount = () => console.log('navigateToAccount');

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AccountDropdown
        t={t}
        paths={paths}
        currentPathname={pathname}
        username={username}
        logout={logout}
        navigateToAccount={navigateToAccount}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
