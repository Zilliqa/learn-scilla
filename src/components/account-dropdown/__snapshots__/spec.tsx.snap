// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Dropdown tests basic tests matches the snapshot 1`] = `
<li
  className="dropdown nav-item"
  data-test-id="account-dropdown"
  onKeyDown={[Function]}
>
  <a
    aria-expanded={false}
    aria-haspopup={true}
    className="active dropdown-toggle nav-link"
    href="#"
    onClick={[Function]}
  >
    noel
  </a>
  <div
    aria-hidden={true}
    className="dropdown-menu dropdown-menu-right"
    role="menu"
    size="sm"
    tabIndex="-1"
  >
    <button
      aria-label="account.account"
      className="btn btn-md type-tertiary btn-block text-left"
      onClick={[Function]}
      type="button"
    >
       
      account.account
       
    </button>
    <button
      aria-label="link.signOut"
      className="btn btn-md type-tertiary btn-block text-left"
      onClick={[Function]}
      type="button"
    >
       
      link.signOut
       
    </button>
  </div>
</li>
`;
